library/core/src/fmt/mod.rs

```diff
@@ -430,6 +438,15 @@ impl<'a> Arguments<'a> {
             _ => None,
         }
     }
+
+    /// Same as [`Arguments::as_str`], but will only return `Some(s)` if it can be determined at compile time.
+    #[must_use]
+    #[inline]
+    const fn as_const_str(&self) -> Option<&'static str> {
+        let s = self.as_str();
+        // SAFETY: both cases are valid as the result
```

Comment:
The last sentence in the intrinsic's docs:
> When calling this in a const fn, both paths must be semantically equivalent, that is, the result of the true branch and the false branch must return the same value and have the same side-effects no matter what.

You're not meeting that requirement here, with maybe-`Some` vs. always-`None`. Do you still get the optimizations you want if this is not a `const fn`?

Reply:
I just tried it without the `const`, and it produced the same result